source/common/router/config_impl.cc

```diff
@@ -1101,7 +1102,30 @@ RouteConstSharedPtr RouteEntryImplBase::clusterEntry(const Http::HeaderMap& head
 RouteConstSharedPtr RouteEntryImplBase::pickWeightedCluster(const Http::HeaderMap& headers,
                                                             const uint64_t random_value,
                                                             const bool ignore_overflow) const {
-  const uint64_t selected_value = random_value % total_cluster_weight_;
+  absl::optional<uint64_t> random_value_from_header;
+  // Retrieve the random value from the header if corresponding header name is specified.
+  if (!random_value_name_.empty()) {
```

Comment:
nit: rename this so it's clear that it indicates a header name?

Reply:
Thanks for the review!
I feel `header_name` here is a bit ambiguous. So I rename it to `random_value_header_name` to make it more explicit.